src/libraries/System.Private.CoreLib/src/System/Numerics/Vector_1.cs

```diff
@@ -0,0 +1,1507 @@
+using System.Text;
+
+using Internal.Runtime.CompilerServices;
+
+namespace System.Numerics
+{
+    /* Note: The following patterns are used throughout the code here and are described here
+    *
+    * PATTERN:
+    *    if (typeof(T) == typeof(int)) { ... }
+    *    else if (typeof(T) == typeof(float)) { ... }
+    * EXPLANATION:
+    *    At runtime, each instantiation of Vector<T> will be type-specific, and each of these typeof blocks will be eliminated,
+    *    as typeof(T) is a (JIT) compile-time constant for each instantiation. This design was chosen to eliminate any overhead from
+    *    delegates and other patterns.
+    *
+    * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+    /// <summary>
+    /// A structure that represents a single Vector. The count of this Vector is fixed but CPU register dependent.
+    /// This struct only supports numerical types. This type is intended to be used as a building block for vectorizing
+    /// large algorithms. This type is immutable, individual elements cannot be modified.
+    /// </summary>
+    [Intrinsic]
+    public struct Vector<T> : IEquatable<Vector<T>>, IFormattable where T : struct
+    {
+        private ulong _00;
+        private ulong _01;
+
+        /// <summary>Returns the number of elements stored in the vector. This value is hardware dependent.</summary>
+        public static int Count
+        {
+            [Intrinsic]
+            get
+            {
+                ThrowHelper.ThrowForUnsupportedVectorBaseType<T>();
+                return Unsafe.SizeOf<Vector<T>>() / Unsafe.SizeOf<T>();
+            }
+        }
+
+        /// <summary>Returns a vector containing all zeroes.</summary>
+        public static Vector<T> Zero
+        {
+            [Intrinsic]
+            get
+            {
+                ThrowHelper.ThrowForUnsupportedVectorBaseType<T>();
+                return default;
+            }
+        }
+
+        /// <summary>Returns a vector containing all ones.</summary>
+        public static Vector<T> One
+        {
+            [Intrinsic]
+            get
+            {
+                ThrowHelper.ThrowForUnsupportedVectorBaseType<T>();
+                return new Vector<T>(GetOneValue());
+            }
+        }
+
+        internal static Vector<T> AllBitsSet
+        {
+            [Intrinsic]
+            get
+            {
+                ThrowHelper.ThrowForUnsupportedVectorBaseType<T>();
+                return new Vector<T>(GetAllBitsSetValue());
+            }
+        }
+
+        /// <summary>Constructs a vector whose components are all <paramref name="value" />.</summary>
+        [Intrinsic]
+        public unsafe Vector(T value) : this()
+        {
+            ThrowHelper.ThrowForUnsupportedVectorBaseType<T>();
+
+            for (nint index = 0; index < Count; index++)
+            {
+                SetElement(index, value);
```

Comment:
The logic which uses `Unsafe.Add(Unsafe.As())` is relegated to `GetElement` and `SetElement`.